packages/@react-aria/virtualizer/src/ScrollView.tsx

```diff
@@ -128,8 +129,9 @@ function ScrollView(props: ScrollViewProps, ref: RefObject<HTMLDivElement>) {
       return;
     }
 
-    let w = dom.clientWidth;
-    let h = dom.clientHeight;
+    let computedStyles = getComputedStyle(dom);
+    let w = parseFloat(computedStyles.width) + parseFloat(computedStyles.paddingLeft) + parseFloat(computedStyles.paddingRight) || 0;
+    let h = parseFloat(computedStyles.height) + parseFloat(computedStyles.paddingTop) + parseFloat(computedStyles.paddingBottom) || 0;
```

Comment:
recommenting:
As mentioned in https://github.com/adobe/react-spectrum/pull/4846/files#r1279884044, the core issue was that the TableView's column head wrapper would occasionally be wider than the actual TableBody due to clientWidth's pixel rounding behavior. This would result in the window/flex container around the TableView rendering scrollbars if the TableView's width was matching the exact width of said window/flex container -> resulting in the TableView resizing itself to match the new (smaller) available width of the container -> a new smaller visibleRect is calculated and applied to the TableView's header width causing the scrollbars to disappear -> cycle begins again

Reply:
did you try `getBoundingClientRect`? I think that has sub-pixel precession, and then you wouldn't need to manually add the padding and mock as many things.

Reply:
it does have sub-pixel precision but it also includes the border and the scrollbar width which we don't want unfortunately since the TableHeader wrapper's width should only match the ScrollView's width + padding. Doing so results in extra scrolling when scrollbars are enabled:

<img width="486" alt="image" src="https://github.com/adobe/react-spectrum/assets/9661127/16e8946a-9dd5-42de-895b-6d9e17a52084">

What makes it even more troublesome is that the TableHeader wrapper doesn't have `box-sizing: border-box` so we also have to account for the extra 1px applied on either end when calculating what width should be applied to it (aka we need it to match the scrollable body's layout, border and all, so that the column headers line up with the row cells)

Reply:
Doing something like using `width: 100%` for the TableHeader wrapper width instead of using the visible rect + `box-sizing: border-box` and reverting back to using `dom.clientWidth` in scrollView seems to work decently (makes the TableHeader wrapper really wide when we have a lot of rows but doesn't seem to break anything visually), but then results in stuff like missing top border due to element overlap:
<img width="371" alt="image" src="https://github.com/adobe/react-spectrum/assets/9661127/f600b282-52d9-4a4e-8c0f-58bb2c7ccb4a">

fixing the above results in:
<img width="253" alt="image" src="https://github.com/adobe/react-spectrum/assets/9661127/cb091ba4-e656-47bc-b785-1023472ba74c">

which is the original issue @snowystinger ran into all that time ago with resizing columns (aka the break in the blue line). I haven't been able to find a way to fix the above, otherwise I would've went with the `width: 100%` route rather than the computed styles

Reply:
There's an old PR or commit which moved the blue line up and out so it was a single element instead of two. Forget where it is, but you might look for that.

Reply:
Oh thanks for the heads up! It actually turns out the `width: 100%` route was non-viable for other reasons (made the column header too wide in the many rows case) so I think we are good here for now (fingers crossed).